### DISCUSSION TITLE
[06-Face Recognition] max number of faces and spoof detection algorithm

### DISCUSSION THREAD

**Discussion by shingo-kj** [Ignitarium-Renesas/RZV2L_AiLibrary#1] [Q&A]
Thank you for providing many useful libraries.
Let me have three question about '06-Face Recognition and Spoof Detection' library.
Q1 : What is the maximum number of faces that can be registered ?
Q2 : What is the algorithm for spoof detection ?
Q3 : What are some examples of impersonation detection failures?
      For example, when a high quality photo is recognized as a real face.

Thank you in advance and best regards,
  shingo-kj

**Comment by mohitb-ign** [comment]
Please find the answers below:
Answer 1: In current application, the number of faces registered is limited by device's storage memory. 'Face registration' just creates embeddings of a new face and stores in the device. 
When a new facial sample is seen by the application, it tries to match the embeddings of the new face with the existing embeddings. Before releasing the application to GitHub, we have done testing of face recognition with 10-12 unique faces. In case the requirement is to support face recognition for more faces, better deep neural network model and more sophisticated approach in embedding matching is required.

Answer 2 : It is Resnet-50 deep neural network trained as binary classifier. The model is trained with spoof images from LED/LCD screens.

Answer 3: The current implementation is able to detect spoofing done by using photos on LED/LCD screens. It will fail in case of high quality print on the paper or in case of a printed or full face covering face mask.
